File: katalyze/src/messageHandlers/ProblemHandler.java
package messageHandlers;

import org.apache.log4j.Logger;
import model.*;
import io.*;

public class ProblemHandler extends SingleMessageHandler {

	static Logger logger = Logger.getLogger(ProblemHandler.class);
	
	public ProblemHandler() { super("problem"); }
		
	public void process(SimpleMessage message) {
		int id = message.getInt("id");
		String abbrev = ""+Character.toChars(64+id)[0];
		String problemName = message.get("name").trim();
	    logger.info("addProblem(" + message.get("id") + ", " + id + ", " + abbrev + ", " + problemName + ")");
	    Problem newProblem = new Problem(id, problemName);
		contest.addProblem(newProblem);
	}

}

File: katalyze/src/main/kotlin/messageHandlers/ProblemHandler.kt
package messageHandlers

import legacyfeed.SimpleMessage
import model.Problem
import org.apache.logging.log4j.kotlin.logger

class ProblemHandler : SingleMessageHandler("problem") {
    override fun process(message: SimpleMessage) {
        val id = message["id"]

        val abbrev = getProblemAbbreviation(id)

        val problemName = message["name"].trim { it <= ' ' }
        logger.info("addProblem(" + message["id"] + ", " + id + ", " + abbrev + ", " + problemName + ")")
        val newProblem = Problem(id, problemName, abbrev, null)
        contest!!.addProblem(newProblem)
    }

    companion object {
        private val logger = logger()

        private fun getProblemAbbreviation(id: String): String {
            try {
                val idAsInteger = id.toInt()
                return "" + Character.toChars(64 + idAsInteger)[0]
            } catch (e: NumberFormatException) {
                return id
            }
        }
    }
}